crates/op-steel/src/optimism/mod.rs

```diff
@@ -171,17 +173,37 @@ impl EvmBlockHeader for OpBlockHeader {
     fn to_block_env(&self, spec: OpSpecId) -> BlockEnv {
         let header = self.0.inner();
 
+        let blob_excess_gas_and_price =
+            header
+                .excess_blob_gas
+                .map(|excess_blob_gas| match spec.into_eth_spec() {
+                    SpecId::CANCUN => BlobExcessGasAndPrice::new(
+                        excess_blob_gas,
+                        eip4844::BLOB_GASPRICE_UPDATE_FRACTION as u64,
+                    ),
+                    SpecId::PRAGUE => BlobExcessGasAndPrice::new(
+                        excess_blob_gas,
+                        eip7691::BLOB_GASPRICE_UPDATE_FRACTION_PECTRA as u64,
+                    ),
+                    SpecId::OSAKA => BlobExcessGasAndPrice::new(
+                        excess_blob_gas,
+                        eip7691::BLOB_GASPRICE_UPDATE_FRACTION_PECTRA as u64,
+                    ),
+                    _ => unimplemented!(
+                        "unsupported spec with `excess_blob_gas`: {}",
+                        <&'static str>::from(spec)
+                    ),
+                });
```

Comment:
I ran into this with a bit of my work. Can we have a public utility function for this, or do you think that would be a bad idea?